metadata-ingestion/src/datahub/ingestion/source/usage/usage_common.py

```diff
@@ -90,4 +111,20 @@ def make_usage_workunit(
 
 
 class BaseUsageConfig(BaseTimeWindowConfig):
-    top_n_queries: Optional[pydantic.PositiveInt] = 10
+    top_n_queries: pydantic.PositiveInt = 10
+
+    @pydantic.validator("top_n_queries")
+    def ensure_top_n_queries_is_not_too_big(cls, v: int) -> int:
+        minimum_query_size = 20
+
+        max_queries = int(
+            GenericAggregatedDataset.total_budget_for_query_list / minimum_query_size
+        )
+        if (
+            int(GenericAggregatedDataset.total_budget_for_query_list / v)
+            < minimum_query_size
+        ):
+            raise ValueError(
+                f"top_n_queries is set to {v} but it can be maximum {max_queries}"
```

Comment:
in what cases would this error be raised vs the one above?

Reply:
This validated config property `top_n_queries` by calculating max query size by dividing the budget with top_n_queries. 
If this number gets too low (20 chars) then it throws a validation exception and asking to decrease the top_n_queries to make sure the trimmed query is not too small.